controllers/reconcile_test.go

```diff
@@ -73,14 +73,14 @@ var (
 type check[T any] func(t *testing.T, params T)
 
 func TestOpenTelemetryCollectorReconciler_Reconcile(t *testing.T) {
```

Comment:
can you add a test in here for the rbac specific finalizer logic?

Reply:
Added a test